runtime/shared_common/shrinit.cpp

```diff
@@ -4923,4 +4931,201 @@ isFreeDiskSpaceLow(J9JavaVM *vm, U_64* maxsize)
+		 if (NULL == key) {
+			 Trc_SHR_INIT_storeStartupHintsToSharedCache_Null_key(currentThread);
+			 return ret;
+		 } else {
+			 vm->sharedClassConfig->localStartupHints.key = key;
+		 }
+	}
+	if (J9_ARE_ANY_BITS_SET(vm->sharedClassConfig->localStartupHints.localStartupHintFlags, J9SHR_LOCAL_STARUPHINTS_FLAG_WRITE_HINTS)) {
+		J9SharedDataDescriptor dataDescriptor = {0};
+		UDATA flag = J9SHRDATA_SINGLE_STORE_FOR_KEY_TYPE;
+		dataDescriptor.address = (U_8*)&vm->sharedClassConfig->localStartupHints.hintsData;
+		dataDescriptor.type = J9SHR_DATA_TYPE_STARTUP_HINTS;
+		dataDescriptor.length = sizeof(J9SharedStartupHintsDataDescriptor);
+		if (J9_ARE_ANY_BITS_SET(vm->sharedClassConfig->localStartupHints.localStartupHintFlags, J9SHR_LOCAL_STARUPHINTS_FLAG_OVERWRITE_HINTS)) {
+			flag = J9SHRDATA_SINGLE_STORE_FOR_KEY_TYPE_OVERWRITE;
+		}
+		dataDescriptor.flags = flag;
+		Trc_SHR_INIT_storeStartupHintsToSharedCache_Set_Flags(currentThread, flag);
+		ret = j9shr_storeSharedData(currentThread, key, strlen(key), &dataDescriptor);
+		if (NULL == ret) {
+			Trc_SHR_INIT_storeStartupHintsToSharedCache_Store_Failed(currentThread);
+		} else {
+			Trc_SHR_INIT_storeStartupHintsToSharedCache_Store_Successful(currentThread, vm->sharedClassConfig->localStartupHints.hintsData.flags,
+					vm->sharedClassConfig->localStartupHints.hintsData.heapSize1, vm->sharedClassConfig->localStartupHints.hintsData.heapSize2);
+		}
+	} else {
+		Trc_SHR_INIT_storeStartupHintsToSharedCache_Store_Nothing(currentThread);
+	}
+	return ret;
+}
+
+/**
+ * Stores the GC hints into vm->sharedClassConfig->localStartupHints.hintsData. This function is not thread safe.
+ * @param[in] vmThread  The current thread
+ * @param[in] heapSize1  The first heap size param that is to be stored into the shared cache
+ * @param[in] heapSize2  The second heap size param that is to be stored into the shared cache
+ * @param[in] forceReplace TRUE Replace the existing GC hints under the same key (the same command line arguments) if there is one already in the shared cache.
+ * 							FALSE Do not replace existing GC hints under the same key.
+ */
+void
+j9shr_storeGCHints(J9VMThread* currentThread, UDATA heapSize1, UDATA heapSize2, BOOLEAN forceReplace)
+{
+	J9JavaVM* vm = currentThread->javaVM;
+
+	if (TRUE == forceReplace) {
+		vm->sharedClassConfig->localStartupHints.hintsData.heapSize1 = heapSize1;
+		vm->sharedClassConfig->localStartupHints.hintsData.heapSize2 = heapSize2;
+		vm->sharedClassConfig->localStartupHints.hintsData.flags |= J9SHR_STARUPHINTS_HEAPSIZES_SET;
+		vm->sharedClassConfig->localStartupHints.localStartupHintFlags |= J9SHR_LOCAL_STARUPHINTS_FLAG_OVERWRITE_HEAPSIZES;
+		Trc_SHR_INIT_j9shr_storeGCHints_Overwrite_LocalHints(currentThread, heapSize1, heapSize2);
+	} else {
+		if (J9_ARE_NO_BITS_SET(vm->sharedClassConfig->localStartupHints.hintsData.flags, J9SHR_STARUPHINTS_HEAPSIZES_SET)) {
+			vm->sharedClassConfig->localStartupHints.hintsData.heapSize1 = heapSize1;
```

Comment:
These 3 lines are the same as the forceReplace case, I think this can be restructured to avoid the duplication.

Reply:
Fixed